datavec/datavec-python/src/main/java/org/datavec/python/NumpyArray.java

```diff
@@ -0,0 +1,179 @@
+    private long[] strides;
+    private DataType dtype = DataType.FLOAT;
+    private INDArray nd4jArray;
+
+    public NumpyArray(long address, long[] shape, long strides[], boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+
+        }
+    }
+    public NumpyArray(long address, long[] shape, long strides[]){
+        this(address, shape, strides, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype){
+        this(address, shape, strides, dtype, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype, boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        this.dtype = dtype;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+        }
+    }
+
+    public long getAddress() {
+        return address;
+    }
+
+    public long[] getShape() {
+        return shape;
+    }
+
+    public long[] getStrides() {
+        return strides;
+    }
+
+    public DataType getDType() {
+        return dtype;
+    }
+
+    public JSONObject toJSON(){
+        JSONObject jsonObject = new JSONObject();
+        JSONArray data = new JSONArray();
+        long size = 1;
+        for (long d: nd4jArray.shape()){
+            size *= d;
+        }
+        if (dtype == DataType.SHORT || dtype == DataType.INT || dtype == DataType.LONG){
+            for (int i=0; i < size; i++){
+                data.add((long)nd4jArray.getDouble(i));
+            }
+        }
+        else if (dtype == DataType.FLOAT || dtype == DataType.DOUBLE){
+            for (int i=0; i<size; i++){
+                data.add(nd4jArray.getDouble(i));
+            }
+        }
+        JSONArray shape = new JSONArray();
+        for (long d: this.shape){
+            shape.add(d);
+        }
+        String dtypeStr = dtype.toString();
+        jsonObject.put("data", data);
+        jsonObject.put("shape", shape);
+        jsonObject.put("dtype", dtypeStr);
+        return jsonObject;
+    }
+
+
+    public NumpyArray(JSONObject json) throws Exception{
+        address = (Long)json.get("address");
+        JSONArray shapeJson = (JSONArray)json.get("shape");
+        shape = new long[shapeJson.size()];
+        for (int i=0; i<shape.length; i++){
+            shape[i] = (Long)shapeJson.get(i);
+        }
+        JSONArray stridesJson = (JSONArray)json.get("strides");
+        strides = new long[stridesJson.size()];
+        for (int i=0; i<strides.length; i++){
+            strides[i] = (Long)stridesJson.get(i);
+        }
+        String dtypeStr = (String)json.get("dtype");
+        if (dtypeStr != null){
+            if (dtypeStr.equals("FLOAT")){
```

Comment:
Why not just use Dtype.fromString(..)?

Reply:
`DataType.fromString()` does not exist. Want me to add it to Nd4j?

Reply:
We're supposed to have string support now, but not sure it's all fully there yet, @raver119?

Reply:
will fix after string support added to nd4j